b2share/modules/upgrade/upgrades/upgrade_2_0_0_to_2_1_0.py

```diff
@@ -157,6 +159,14 @@ def alembic_upgrade_database_data(alembic, verbose):
 
 def migrate_record_metadata(record, parent_pid):
     """Migrate a record's metadata to 2.1.0 format."""
+    # Fix bad OAI-PMH identifiers
+    oai_prefix = current_app.config.get('OAISERVER_ID_PREFIX')
+    if oai_prefix:
+        oai_id = record.get('_oai', {}).get('id')
+        if oai_id and not oai_id.startswith(oai_prefix):
+            assert oai_id == record['_deposit']['id']
+            record['_oai']['id'] = oai_prefix + oai_id
```

Comment:
The following code only fixes the "minted" OAI ID which is in the record. Are we sure that the PersistentIdentifier is correct?
You can see the [following line](https://github.com/EUDAT-B2SHARE/b2share/blob/bee5bbab68e8f9fffa47988355a7088e04585e00/b2share/modules/records/minters.py#L73) in our minter. I suspect that if we minted the wrong PIDs, they are also wring in the PersistentIdentifier table.

Reply:
The PIDs are correct, you can see here that we're using the right value:
https://github.com/EUDAT-B2SHARE/b2share/blob/bee5bbab68e8f9fffa47988355a7088e04585e00/b2share/modules/records/minters.py#L76